### PR TITLE
Tab completion for /format values.

### PR DESCRIPTION
**Pull request by LemonBoy**
I've tested this very briefly and it does seem to be consistent with how the completion for the other commands behaves.

### CODE REVIEW

**Comment by vague666** [comment]
Good work \m/ :)

File: src/fe-common/core/themes.c [hunk 1]
```diff
@@ -1289,12 +1290,12 @@ static GList *completion_get_formats(const char *module, const char *key)
 	list = NULL;
 
 	modules = get_sorted_modules();
-	if (*module == '\0' || theme_search(modules, module) != NULL) {
+	if (module == NULL || theme_search(modules, module) != NULL) {
 		for (tmp = modules; tmp != NULL; tmp = tmp->next) {
 			THEME_SEARCH_REC *rec = tmp->data;
 
-			if (*module == '\0' || g_ascii_strcasecmp(rec->short_name, module) == 0)
```
Comment (ailin-nemui) [inline src/fe-common/core/themes.c]: question from ahf what motivates this change?

Comment (LemonBoy) [inline src/fe-common/core/themes.c]: I do not like throwing around an empty string when `NULL` could do just fine.

**Comment by ailin-nemui** [comment]
`/format irc <TAB>` no longer completes to `netsplit`, `netsplit_more`, ... `away`...

**Comment by LemonBoy** [comment]
The easy fix would be to check whether `irc` matches a module name but I'd prefer to change the format accepted by the command to a IMO saner `/format [<module>/]<name>`.

**Comment by ailin-nemui** [comment]
the old code somehow did that.. but I really like having completion for format values, too! I am also afraid that some private script may be using /format module name, in general would be nice if we can keep the syntax imo.. although I do not feel very strongly

**Comment by LemonBoy** [comment]
I don't care enough about this to keep the PR open, if anyone wants to take care of this should feel free to pick this up.

**Pull request closed without merging**